bin/update_bitmasks

```diff
@@ -0,0 +1,326 @@
+particular redshift fit or associated QA.
+ZWARN==0 is good; any non-zero value indicates a potential problem.
+This mask will be described in more detail in the Redrock paper
+(Bailey et al 2023 in prep), as well as Section 5.3.1 of the
+Survey Operations Paper (Schlafly et al 2023 TODO: add link).
+
+The canonical code location defining these bits is
+`desitarget targetmask.yaml <https://github.com/desihub/desitarget/blob/main/py/desitarget/data/targetmask.yaml#L230>`_.
+Bits 0-15 are set by Redrock itself (the redshift fitter),
+while bits 16-19 are set by DESI-specific post-processing.
+
+**ZWARN Mask Locations**
+
+================ ============= ===========
+File             Table HDU     Column
+================ ============= ===========
+`redrock`_       REDSHIFTS     ZWARN
+`emline`_        EMLINEFIT     ZWARN
+`zmtl`_          ZMTL          ZWARN
+`mtl`_           MTL           ZWARN
+`zpix`_          ZCATALOG      ZWARN
+`ztile`_         ZCATALOG      ZWARN
+`zall`_          ZCATALOG      ZWARN
+`lss`_           LSS           ZWARN
+================ ============= ===========
+
+.. _`redrock`: DESI_SPECTRO_REDUX/SPECPROD/tiles/GROUPTYPE/TILEID/GROUPID/redrock-SPECTROGRAPH-TILEID-GROUPID.html
+.. _`emline`: DESI_SPECTRO_REDUX/SPECPROD/tiles/GROUPTYPE/TILEID/GROUPID/emline-SPECTROGRAPH-TILEID-GROUPID.html
+.. _`zmtl`: DESI_SPECTRO_REDUX/SPECPROD/tiles/GROUPTYPE/TILEID/GROUPID/zmtl-SPECTROGRAPH-TILEID-GROUPID.html
+.. _`mtl`: DESI_SURVEYOPS/mtl/main/dark/mtl-dark-hp-HPX.html
+.. _`zpix`: DESI_SPECTRO_REDUX/SPECPROD/zcatalog/zpix-SURVEY-PROGRAM.html
+.. _`ztile`: DESI_SPECTRO_REDUX/SPECPROD/zcatalog/ztile-SURVEY-PROGRAM-GROUPTYPE.html
+.. _`zall`: DESI_SPECTRO_REDUX/SPECPROD/zcatalog/zall-pix-SPECPROD.html
+.. _`lss`: DESI_ROOT/vac/RELEASE/lss/VERSION/LSScats/full/fullVETO_dat.html
+
+**ZWARN Bit Definitions**
+
+{ZWARN}
+
+Spectroscopic Reduction Masks
+-----------------------------
+
+The **FIBERSTATUS** bit mask records the state of individual
+fibers for issues that impact the entire spectrum, e.g. a broken fiber.
+The **SPECMASK** bit mask tracks wavelength dependent isses per spectrum,
+e.g. masks for cosmic rays.
+
+FIBERSTATUS
+~~~~~~~~~~~
+
+The **FIBERSTATUS** mask is kept as a column in FIBERMAP and related HDUs.
+Bits 0-7 are set by fiber assignment from focal plane information known
+before observations; bits 8-24 are set by the spectroscopic pipeline;
+bits 25-30 are set by the final QA step to set bits for all fibers in
+a petal (e.g. because sky model noise makes all spectra questionable).
+
+The canonical code location defining FIBERSTATUS bits is
+`desispec.maskbits L55 <https://github.com/desihub/desispec/blob/0.55.0/py/desispec/maskbits.py#L55>`_.
+
+**FIBERSTATUS Mask Locations**
+
+================ ============= ===========
+File             Table HDU     Column
+================ ============= ===========
+`frame`_         FIBERMAP      FIBERSTATUS
```

Comment:
OK, I see what you're trying to do here.

Just embed the `:doc:` link in the table. Don't bother trying to make it act like a footnote.

Reply:
I didn't embed the `:doc:` link in the table because that would make the column super wide and hard to read in the python/rst files when editing.  Using the footnote style of the link keeps the raw table readable.

Reply:
footnote forms that I tried to avoid the "html" that don't work:
```
.. _:doc:`frame <DESI_SPECTRO_REDUX/SPECPROD/exposures/NIGHT/EXPID/frame-CAMERA-EXPID>`
.. _`frame <DESI_SPECTRO_REDUX/SPECPROD/exposures/NIGHT/EXPID/frame-CAMERA-EXPID>`
--> both result in "WARNING: malformed hyperlink target"

.. _`frame`: :doc:`<DESI_SPECTRO_REDUX/SPECPROD/exposures/NIGHT/EXPID/frame-CAMERA-EXPID>`
.. _`frame`: :doc:`frame <DESI_SPECTRO_REDUX/SPECPROD/exposures/NIGHT/EXPID/frame-CAMERA-EXPID>`
--> both have no sphinx errors or warnings, but generated link is wrong (includes the ":doc:" in the link itself, as well as "%60%3C" .
```
In the end this works for both getting the correct link and not having sphinx/rst errors:
```
.. _`frame`: DESI_SPECTRO_REDUX/SPECPROD/exposures/NIGHT/EXPID/frame-CAMERA-EXPID.html
```
It's a bit surprising that requires including "html" directly in the rst text, but ok.

Reply:
OK, what we needed here was *substitutions*. Please take a look.